src/supertux/error_handler.cpp

```diff
@@ -49,33 +193,173 @@ ErrorHandler::handle_error(int sig)
+      2, // buttonid
+      "OK" // text
+    }
+  };
+
+  SDL_MessageBoxData data = {
+    SDL_MESSAGEBOX_ERROR, // flags
+    nullptr, // window
+    "Error", // title
+    msg, // message
+    SDL_arraysize(btns), // numbuttons
+    btns, // buttons
+    nullptr // colorscheme
+  };
 
-  // Print out all the frames to stderr.
-  backtrace_symbols_fd(array, static_cast<int>(size), STDERR_FILENO);
+  int resultbtn;
+  SDL_ShowMessageBox(&data, &resultbtn);
+
+  switch (resultbtn)
+  {
+    case 0:
+      report_error(stacktrace);
+      break;
+
+    case 1:
+    {
+      SDL_MessageBoxButtonData detailsbtns[] = {
+        {
+          0, // flags
+          0, // buttonid
+          "Report" // text
+        },
+        {
+          SDL_MESSAGEBOX_BUTTON_RETURNKEY_DEFAULT, // flags
+          1, // buttonid
+          "OK" // text
+        }
+      };
+
+      data = {
+        SDL_MESSAGEBOX_ERROR, // flags
+        nullptr, // window
+        "Error details", // title
+        stacktrace.c_str(), // message
+        SDL_arraysize(detailsbtns), // numbuttons
+        detailsbtns, // buttons
+        nullptr // colorscheme
+      };
+
+      SDL_ShowMessageBox(&data, &resultbtn);
+
+      if (resultbtn == 0)
+        report_error(stacktrace);
+
+      break;
+    }
+
+    default:
+      break;
+  }
+}
+
+void
+ErrorHandler::error_dialog_exception(const std::string& exception)
+{
+  std::stringstream stream;
+
+  stream << "SuperTux has encountered a fatal exception!";
+
+  if (!exception.empty())
+  {
+    stream << "\n\n" << exception;
+  }
+
+  std::string msg = stream.str();
+
+  SDL_MessageBoxButtonData btns[] = {
+#ifdef WIN32
+    {
+      0, // flags
+      0, // buttonid
+      "Open log" // text
+    },
+#endif
+    {
+      SDL_MESSAGEBOX_BUTTON_RETURNKEY_DEFAULT, // flags
+      1, // buttonid
+      "OK" // text
+    }
+  };
+
+  SDL_MessageBoxData data = {
+    SDL_MESSAGEBOX_ERROR, // flags
+    nullptr, // window
+    "Error", // title
+    msg.c_str(), // message
+    SDL_arraysize(btns), // numbuttons
+    btns, // buttons
+    nullptr // colorscheme
+  };
+
+  int resultbtn;
+  SDL_ShowMessageBox(&data, &resultbtn);
+
+#ifdef WIN32
+  if (resultbtn == 0)
+  {
+    // Repurpose the stream.
+    stream.str("");
+
+    stream << SDL_GetPrefPath("SuperTux", "supertux2")
+           << "/console.err";
+    FileSystem::open_path(stream.str());
+  }
 #endif
 }
 
+void
+ErrorHandler::report_error(const std::string& details)
+{
+  std::stringstream url;
+
+  // cppcheck-suppress unknownMacro
+  url << "https://github.com/supertux/supertux/issues/new"
+         "?template=crash.yml"
+         "&labels=type:crash,status:needs-confirmation"
+         "&supertux-version=" << FileSystem::escape_url(PACKAGE_VERSION) <<
+         "&system-info=" << FileSystem::escape_url(get_system_info()) <<
+         "&debug-stacktrace=" << FileSystem::escape_url(details);
+
+  FileSystem::open_url(url.str());
```

Comment:
You should escape the entire string instead of the individual strings

Reply:
I guess that's also a way.

Reply:
Wouldn't that lead to the ampersands getting escaped as well?

Reply:
true. Nvm